cli: improve the debug pebble db excise command

RaduBerinde opened the pull request:
This commit makes the recently added `pebble db excise` command more
user-friendly: we allow excising an entire table by specifying the
table and tenant IDs. Example:

```
./cockroach debug pebble db excise cockroach-data --table-id=4 --tenant-id=2
Excising range:
  start: /Tenant/2/Table/4/0,0
  end:   /Tenant/2/Table/5/0,0
WARNING!!!
This command will remove all keys in this range!
Continue? [Y/N] y
Excise complete.
```

Fixes: #122059
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/122567)
<!-- Reviewable:end -->

itsbilal commented on pkg/cli/debug.go:
```diff
@@ -1631,10 +1634,37 @@ func initPebbleCmds(cmd *cobra.Command, pebbleTool *tool.T) {
+	if debugPebbleOpts.exciseTenantID == 0 && debugPebbleOpts.exciseTableID == 0 {
+		return pebble.KeyRange{}, nil
+	}
+	if debugPebbleOpts.exciseTenantID == 0 || debugPebbleOpts.exciseTableID == 0 {
```
There might also be value in blowing up the whole tenant, right? in which case we should be able to specify just the tenant ID and not the table ID?

RaduBerinde replied:
CC @msbutler WDYT?

msbutler replied:
yeah, this could be nice in a UA cluster, if we needed to wipe the main tenant.

RaduBerinde replied:
Done. I added an `--entire-tenant` flag which must be used in conjunction with `--tenant-id` (to make accidents less likely). We disallow wiping the entire system tenant.

itsbilal reviewed (approved):
LGTM

RaduBerinde committed:
cli: improve the debug pebble db excise command

This commit makes the recently added `pebble db excise` command more
user-friendly: we allow excising an entire table or tenant. Example:

```
./cockroach debug pebble db excise cockroach-data --table-id=4 --tenant-id=2
Excising range:
  start: /Tenant/2/Table/4/0,0
  end:   /Tenant/2/Table/5/0,0
WARNING!!!
This command will remove all keys in this range!
Continue? [Y/N] y
Excise complete.
```

Fixes: #122059
Release note: None

itsbilal reviewed (approved):
LGTM

RaduBerinde commented:
TFTR!

bors r+

RaduBerinde commented:
bors r+
